flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSelectiveSpillingStrategyTest.java

```diff
@@ -0,0 +1,163 @@
+                new HsSelectiveSpillingStrategy(
+                        HybridShuffleConfiguration.builder(NUM_SUBPARTITIONS, 1)
+                                .setSelectiveStrategySpillThreshold(SELECTIVE_SPILL_THRESHOLD)
+                                .setSelectiveStrategySpillBufferRatio(SELECTIVE_SPILL_BUFFER_RATIO)
+                                .build());
+    }
+
+    @Test
+    void testOnBufferFinished() {
+        BufferWithIdentity bufferWithIdentity = new BufferWithIdentity(createBuffer(), 0, 0);
+        Action finishedAction =
+                spillStrategy.onBufferFinished(spillInfoProvider, bufferWithIdentity);
+        assertThat(finishedAction.isImmediate()).isTrue();
+        assertThat(finishedAction.getBufferToRelease()).isEmpty();
+        assertThat(finishedAction.getBufferToSpill()).isEmpty();
+    }
+
+    @Test
+    void testOnBufferConsumed() {
+        BufferWithIdentity bufferWithIdentity = new BufferWithIdentity(createBuffer(), 0, 0);
+        Action consumedAction =
+                spillStrategy.onBufferConsumed(spillInfoProvider, bufferWithIdentity);
+        assertThat(consumedAction.isImmediate()).isTrue();
+        assertThat(consumedAction.getBufferToRelease()).hasSize(1);
+        assertThat(consumedAction.getBufferToRelease()).element(0).isEqualTo(bufferWithIdentity);
+        assertThat(consumedAction.getBufferToSpill()).isEmpty();
+    }
+
+    @Test
+    void testOnUsedMemoryChanged() {
+        testOnUsedMemoryLow();
+        testOnUsedMemoryHigh();
+    }
+
+    void testOnUsedMemoryLow() {
+        final int bufferPoolSize = 10;
+        final int bufferThreshold = (int) (bufferPoolSize * SELECTIVE_SPILL_THRESHOLD);
+        spillInfoProvider.setCurrentPoolSizeSupplier(() -> bufferPoolSize);
+        spillInfoProvider.setNumTotalUnSpilledBuffersSupplier(() -> bufferThreshold - 1);
+        Action action = spillStrategy.onUsedMemoryChanged(spillInfoProvider);
+        assertThat(action.isImmediate()).isTrue();
+        assertThat(action.getBufferToSpill()).isEmpty();
+        assertThat(action.getBufferToRelease()).isEmpty();
+    }
+
+    void testOnUsedMemoryHigh() {
+        final int subpartition1 = 0;
+        final int subpartition2 = 1;
+        final int subpartition3 = 2;
+
+        Map<Integer, Deque<BufferWithIdentity>> subpartitionBuffers = new HashMap<>();
+        subpartitionBuffers.put(subpartition1, createBufferWithIdentities(subpartition1, 0, 4, 6));
+        subpartitionBuffers.put(subpartition2, createBufferWithIdentities(subpartition2, 3, 7, 9));
+        subpartitionBuffers.put(subpartition3, createBufferWithIdentities(subpartition3, 5, 6, 11));
+
+        spillInfoProvider.setUnSpilledBuffersFunction(subpartitionBuffers::get);
+        spillInfoProvider.setConsumptionProgressSupplier(() -> Arrays.asList(-1, 1, 2));
+
+        final int totalBuffers = 9;
+        final int expectedNumSpilled = (int) (totalBuffers * SELECTIVE_SPILL_BUFFER_RATIO);
```

Comment:
Shouldn't this be `poolSize * SELECTIVE_SPILL_BUFFER_RATIO`?

Reply:
Yes, it should be more reasonable to use `poolSize` instead of `totalBufferSize`, I will also modify the calculation logic in `HsSelectiveSpillingStrategy`